src/formatter.ts

```diff
@@ -0,0 +1,187 @@
+    console.log("Loaded Brittanies");
+    workspace.onDidChangeConfiguration(
+      (evt) => {
+        if (evt.affectsConfiguration("brittany")) {
+          this.loadSettings();
+        }
+      },
+      this,
+      this.disposables
+    );
+  }
+
+  public dispose(): void {
+    this.disposables.forEach((i) => i.dispose());
+  }
+
+  public async provideDocumentFormattingEdits(
+    document: TextDocument,
+    options: FormattingOptions,
+    token: CancellationToken
+  ): Promise<TextEdit[]> {
+    console.log(`Formatting`);
+    return await this.provideDocumentRangeFormattingEdits(
+      document,
+      null,
+      options,
+      token
+    );
+  }
+
+  public async provideDocumentRangeFormattingEdits(
+    document: TextDocument,
+    range: Range | null,
+    options: FormattingOptions,
+    token: CancellationToken
+  ): Promise<TextEdit[]> {
+    console.log(
+      `Formatting with brittany: ${JSON.stringify({
+        document,
+        options,
+        range,
+        token
+      })}`
+    );
+    const edits: TextEdit[] = [];
+    if (!this.isEnabled) {
+      throw new Error("Brittany formatter not enabled");
+    }
+    // if document has CRLF line endings, change it to LF.
+    if (!this.keepCRLF && document.eol === EndOfLine.CRLF) {
+      console.log("changing line endings to LF");
+      edits.push(TextEdit.setEndOfLine(EndOfLine.LF));
+    }
+
+    const fullRange: Range = this.fullDocumentRange(document);
+    const fullDoc: boolean = !range || range === fullRange;
+    let input: string | TextDocument;
+
+    if (fullDoc) {
+      if (document.isDirty) {
+        input = document.getText();
+      } else {
+        input = document;
+      }
+    } else {
+      input = document.getText(range);
+    }
+
+    const formatted: string = await this.runBrittany(input, document.fileName);
+    edits.push(TextEdit.replace(range || fullRange, formatted));
+    return edits;
+  }
+
+  private loadSettings(): void {
+    const config: any = workspace.getConfiguration("brittany");
+    this.keepCRLF = config.keepCRLF;
+    this.isStackEnabled = config.stackEnable;
+    this.isEnabled = config.enable;
+    this.additionalFlags = config.additionalFlags;
+    this.brittanyCmd = config.path;
+  }
+
+  private async runBrittany(
+    input: string | TextDocument,
+    inputFilename: string
+  ): Promise<string> {
+    let cmdName: string = "brittany";
+    const args: string[] = [];
+    let stdin: string | undefined;
+    let maybeWorkspaceFolder: WorkspaceFolder = null;
+
+    if (this.isStackEnabled) {
+      cmdName = "stack";
+      args.push("exec", "brittany", "--");
+    }
+    if (typeof input === "string") {
+      stdin = input;
+      args.push("--write-mode", "display");
+    } else {
+      args.push(input.uri.fsPath);
+      args.push("--write-mode", "display");
+      maybeWorkspaceFolder = workspace.getWorkspaceFolder(input.uri);
+    }
+
+    const dir: string =
+      maybeWorkspaceFolder !== undefined
```

Comment:
```suggestion
      maybeWorkspaceFolder !== null
```

Should this be `null`?
When `maybeWorkspaceFolder` is declared, it uses a value of `null`. I believe `undefined` is only for uninitialized variables in js, so `maybeWorkspaceFolder` will never be `undefined`

Reply:
Ah, that's right!